src/dex.hs

```diff
@@ -37,6 +40,29 @@ data EvalMode = ReplMode String
 
 data CmdOpts = CmdOpts EvalMode (Maybe FilePath) EvalConfig
 
+
+
+filteredCompletions candidates str = map simpleCompletion $
+  filter (str `isPrefixOf`) candidates
+
+completions :: CompletionFunc (StateT TopEnv IO)
+completions (line, _) = do
+  env <- get
+  let varNames = map (show . pretty) $ envNames env
+  -- note: line and thus word and rest have character order reversed
+  let (word, rest) = break (== ' ') line
+  let anywhereKeywords = ["def", "for", "rof", "case", "data", "where", "of", "if", "then", "else", "interface", "instance", "do", "view"]
+  let startoflineKeywords = ["%bench \"", ":p", ":t", ":html", ":export"]
+  let candidates = varNames ++ anywhereKeywords ++ if null rest
+                                                    then startoflineKeywords
+                                                    else []
```

Comment:
I have no idea what the preferred way to indent this is. 
I hope it isn't this.

Reply:
Well, you could put it on a separate line and make it a one-liner